SocialCareCaseViewerApi.Tests/V1/IntegrationTests/MASH/LinkClientInReferralToResidentInSystem.cs

```diff
@@ -0,0 +1,104 @@
+using NUnit.Framework;
+using SocialCareCaseViewerApi.V1.Boundary.Requests;
+using SocialCareCaseViewerApi.V1.Boundary.Response;
+using SocialCareCaseViewerApi.V1.Infrastructure;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+using MashReferral = SocialCareCaseViewerApi.V1.Infrastructure.MashReferral;
+
+namespace SocialCareCaseViewerApi.Tests.V1.IntegrationTests.MASH
+{
+    [TestFixture]
+    public class LinkClientInReferralToResidentInSystem : IntegrationTestSetup<Startup>
+    {
+        private MashReferral _mashReferral;
+        private MashResident _unlinkedResident;
+        private Person _existingDbPerson;
+
+        [SetUp]
+        public void Setup()
+        {
+            _existingDbPerson = IntegrationTestHelpers.CreateExistingPerson(DatabaseContext);
+            _mashReferral = IntegrationTestHelpers.SaveMashReferralToDatabase(DatabaseContext, "CONTACT");
+
+            _unlinkedResident = IntegrationTestHelpers.CreateUnLinkedMashResident(DatabaseContext, _mashReferral, _existingDbPerson);
+        }
+
+        [Test]
+        public async Task SuccessfulLinkingMatchesMashResidentToSavedPersonInDb()
+        {
+            _unlinkedResident.SocialCareId.Should().BeNull();
+
+            var request = new UpdateMashResidentRequest { SocialCareId = _existingDbPerson.Id };
+            var patchUri = new Uri($"/api/v1/mash-resident/{_unlinkedResident.Id}", UriKind.Relative);
+            var serializedRequest = JsonSerializer.Serialize(request);
+            var requestContent = new StringContent(serializedRequest, Encoding.UTF8, "application/json");
+
+            var linkMashResidentResponse = await Client.PatchAsync(patchUri, requestContent).ConfigureAwait(true);
+
+            linkMashResidentResponse.StatusCode.Should().Be(200);
+
+            _unlinkedResident.SocialCareId.Should().Be(_existingDbPerson.Id);
+
+            var content = await linkMashResidentResponse.Content.ReadAsStringAsync().ConfigureAwait(true);
+            var patchMashResidentResponse = JsonConvert.DeserializeObject<MashResidentResponse>(content);
+
+            patchMashResidentResponse.SocialCareId.Should().Be(_existingDbPerson.Id);
+        }
+
+        [Test]
+        public async Task WhenTheRequestIsInvalidReturnsBadRequestStatusCode()
+        {
+            const string request = "invalid request";
+            var patchUri = new Uri($"/api/v1/mash-resident/{_unlinkedResident.Id}", UriKind.Relative);
+            var serializedRequest = JsonSerializer.Serialize(request);
+            var requestContent = new StringContent(serializedRequest, Encoding.UTF8, "application/json");
+
+            var response = await Client.PatchAsync(patchUri, requestContent).ConfigureAwait(true);
+
+            response.StatusCode.Should().Be(400);
+        }
+
+        [Test]
+        public async Task WhenTheRequestContainsAnIncorrectPersonIdReturnsRelevantErrorMessage()
+        {
+            var incorrectPersonId = _existingDbPerson.Id + 20;
```

Comment:
would the test fail should a person with id +20 actually exist in the db? not sure if it make sense, but would using a negative number id (eg. -400) work for this purpose? happy to chat about it

Reply:
Hi, so this should take the id of the existing person that we set up in the setup stage and add 20 to it. In theory, that ID would not exist (since we've only setup one person in the DB ) and the test database should be wiped before every test. 20 was randomly selected and wasn't sure if there could be issues with possible negative values which is why I decided to add rather than subtract.

Reply:
I  forgot that we're not connecting to the actual db for this 🤦 
makes sense, thank you for explaining 🙇‍♀️

Reply:
The reason I didn't want to randomly generate the ID or give a set value was to avoid the rare chance that the ID we want to test with by coincidence, matched the ID of the person we setup in the database. So by using the existing person's ID as a reference, we should never have the same ID appear.

Reply:
For the integration tests, we should be connecting to the test database in docker but our integration test setup has been designed in way to make sure that we wipe the test database before each integration test is run.